hasher.go

```diff
@@ -46,19 +48,26 @@ func Sha256Namespace8FlaggedLeaf(namespacedData []byte) []byte {
 // The output will also be of length 2*DefaultNamespaceIDLen+sha256.Size = 48 bytes.
 func Sha256Namespace8FlaggedInner(leftRight []byte) []byte {
 	const flagLen = DefaultNamespaceIDLen * 2
-	sha256Len := defaultHasher.Size()
-	left := leftRight[:flagLen+sha256Len]
-	right := leftRight[flagLen+sha256Len:]
+	left := leftRight[:flagLen+sha256.Size]
+	right := leftRight[flagLen+sha256.Size:]
 
 	return defaultHasher.HashNode(left, right)
 }
 
 type Hasher struct {
-	hash.Hash
+	newHashFn    func() hash.Hash
 	NamespaceLen namespace.IDSize
 
 	ignoreMaxNs      bool
 	precomputedMaxNs namespace.ID
+	size             int
+}
+
+func (n *Hasher) Size() int {
```

Comment:
Why is this needed? Hasher(Hash).Size() was called somewhere before? From the codecov bot it seems like this is not called anywhere.

Reply:
It's implictly used by the [`TreeHasher`](https://github.com/lazyledger/nmt/blob/8cd1c48591ee5b84075b1f968628005796f974c6/internal/subtree_hasher.go#L9) which is currently still used by various methods `Prove`,  `ProveNamespace` and maybe more.